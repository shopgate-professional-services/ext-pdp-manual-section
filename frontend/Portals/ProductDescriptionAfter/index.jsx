import React from 'react';
import { RouteContext } from '@shopgate/pwa-common/context';
import { hex2bin } from '@shopgate/pwa-common/helpers/data';
import ManualsSection from '../../components/ManualsSection';
import getConfig from '../../helpers/getConfig';
import connect from './connector';

const config = getConfig();

const ManualsSectionContext = connect(({
  manualUrls,
}) => (
  <ManualsSection
    config={config}
    manualUrls={manualUrls}
  />
));
/**
 * @returns {JSX}
 */
const ProductDescriptionAfter = () => {
  if (config.manualsProperty) {
    return (
      <RouteContext.Consumer>
        {({ params }) =>
          <ManualsSectionContext productId={hex2bin(params.productId)} />
      }
      </RouteContext.Consumer>
    );
  }
  return null;
};

export default ProductDescriptionAfter;

export { ManualsSectionContext as UnwrappedManualsSection };
