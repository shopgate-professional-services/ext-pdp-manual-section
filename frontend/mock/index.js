export const defaultState = {
  product: {
    propertiesByProductId: {
      product_1: {
        properties: [
          {
            label: 'mockone',
            value: 'hi',
          },
          {
            label: 'mocktwo',
            value: 'hello',
          },
        ],
      },
    },
  },
};
export const incorrectPropertiesWithManuals = {
  product: {
    propertiesByProductId: {
      product_1: {
        properties: [
          {
            label: 'Downloads',
            value: 'J.pdf',
          },
          {
            label: 'mocktwo',
            value: 'hello',
          },
        ],
      },
    },
  },
};

export const propertiesWithManuals = {
  product: {
    propertiesByProductId: {
      product_1: {
        properties: [
          {
            label: 'Downloads',
            value: 'B.pdf',
          },
          {
            label: 'mocktwo',
            value: 'hello',
          },
        ],
      },
    },
  },
};

export const mockedConfig = {
  sectionTitle: 'Manuals',
  manualsProperty: 'Downloads',
  manualsPrefixUrl: 'https://example.com/examplePathname',
  typeIndices: [
    {
      searchPattern: 'B.pdf',
      fileType: 'Brochure',
    },
    {
      searchPattern: 'D.pdf',
      fileType: 'Documentation',
    },
  ],
  propertySplit: '\n',
};
