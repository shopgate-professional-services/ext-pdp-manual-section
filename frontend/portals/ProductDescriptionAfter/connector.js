import { connect } from 'react-redux';
import { getManualUrls } from '../../selectors';

/**
 *
 * @param {Object} state The current state.
 * @param {Object} props The props.
 * @returns {Object}
 */
const mapStateToProps = (state, props) => ({
  manualUrls: getManualUrls(state, props),
});

export default connect(mapStateToProps);
