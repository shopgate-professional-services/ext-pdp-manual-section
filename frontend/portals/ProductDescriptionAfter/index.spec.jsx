import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import mockRenderOptions from '@shopgate/pwa-common/helpers/mocks/mockRenderOptions';
import { basicProductState } from '@shopgate/pwa-common-commerce/product/mock';
import { RouteContext } from '@shopgate/pwa-common/context';
import { mockedConfig } from '../../mock';
import ProductDescriptionAfter from './index';

const mockedStore = configureStore();
jest.mock('../../components/ManualsSection', () => () => (<div>ManualsSection</div>));
jest.mock('../../helpers/getConfig', () => () => mockedConfig);

/**
 * Creates component
 * @return {ReactWrapper}
 */
const createComponent = () => mount(
  <RouteContext.Provider value={{ params: { product: 'product_one' } }}>
    <Provider store={mockedStore({ product: basicProductState })}>
      <ProductDescriptionAfter id="foo" />
    </Provider>
  </RouteContext.Provider>,
  mockRenderOptions
);

describe('ManualsSection', () => {
  it('should render with correct config passed', () => {
    const component = createComponent();
    expect(component.find('div').exists()).toBe(true);
    expect(component).toMatchSnapshot();
  });
});
