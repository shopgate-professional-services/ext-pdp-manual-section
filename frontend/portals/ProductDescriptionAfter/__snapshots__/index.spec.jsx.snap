// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManualsSection should render 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ProductDescriptionAfter
    id="foo"
  >
    <Connect(Component)
      productId={false}
    >
      <Component
        dispatch={[Function]}
        manualUrls={Array []}
        productId={false}
      >
        <Component
          config={
            Object {
              "manualsPrefixUrl": "https://example.com/examplePathname",
              "manualsProperty": "Downloads",
              "propertySplit": "
",
              "sectionTitle": "Manuals",
              "typeIndices": Array [
                Object {
                  "fileType": "Brochure",
                  "searchPattern": "B.pdf",
                },
                Object {
                  "fileType": "Documentation",
                  "searchPattern": "D.pdf",
                },
              ],
            }
          }
          manualUrls={Array []}
        >
          <div>
            ManualsSection
          </div>
        </Component>
      </Component>
    </Connect(Component)>
  </ProductDescriptionAfter>
</Provider>
`;
